packages/@ourworldindata/grapher/src/slideInDrawer/SlideInDrawer.tsx

```diff
@@ -0,0 +1,112 @@
+import React from "react"
+import { createPortal } from "react-dom"
+import { computed, action, observable } from "mobx"
+import { observer } from "mobx-react"
+import { GRAPHER_DRAWER_ID } from "../core/GrapherConstants"
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome/index.js"
+import { faXmark } from "@fortawesome/free-solid-svg-icons"
+
+@observer
+export class SlideInDrawer extends React.Component<{
+    title: string
+    active: boolean
+    toggle: () => void
+    children: React.ReactNode
+}> {
+    @observable.ref visible: boolean = this.props.active // true while the drawer is active and during enter/exit transitions
+    contentRef: React.RefObject<HTMLDivElement> = React.createRef()
+
+    componentDidMount(): void {
+        document.addEventListener("keydown", this.onDocumentKeyDown)
+        document.addEventListener("click", this.onDocumentClick, {
+            capture: true,
+        })
+    }
+
+    componentWillUnmount(): void {
+        document.removeEventListener("keydown", this.onDocumentKeyDown)
+        document.removeEventListener("click", this.onDocumentClick)
```

Comment:
This is a bit weird, but we actually need to specify `{capture: true}` here, too - [see MDN](https://developer.mozilla.org/en-US/docs/Web/API/EventTarget/removeEventListener#matching_event_listeners_for_removal).

Reply:
lol I didn't know. good catch!